core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -492,6 +493,30 @@ public HttpResponse handleException(ServiceRequestContext ctx, HttpRequest req,
         }
     }
 
+    private static final class ExceptionHandlingHttpService extends SimpleDecoratingHttpService {
+
+        private final ExceptionHandlerFunction exceptionHandler;
+
+        ExceptionHandlingHttpService(HttpService service, ExceptionHandlerFunction exceptionHandler) {
+            super(service);
+            this.exceptionHandler = exceptionHandler;
+        }
+
+        @Override
+        public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) {
+            try {
+                final HttpResponse response = unwrap().serve(ctx, req);
+                return response.recover(cause -> {
```

Comment:
I'm also wondering if we need to capture this `cause`. Probably not. 😄 
Let's do that if there's a demand for that.

Reply:
We can think of exception handlers attached as domain exception handlers that can fully recover domain exceptions to successful responses.
So I think it is fine as it is.